### PR TITLE
nixos/nix-daemon: add build machine protocol option

### PR DESCRIPTION
**Pull request by yajo**
###### Description of changes

Make ssh-ng:// users happier.

@moduon MT-904

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by bjornfor — APPROVED** [review]

File: nixos/modules/services/misc/nix-daemon.nix [hunk 1]
```diff
@@ -228,6 +228,16 @@ in
                 The hostname of the build machine.
               '';
             };
+            protocol = mkOption {
+              type = types.nullOr types.str;
```
Comment (bjornfor) [inline nixos/modules/services/misc/nix-daemon.nix:232]: On second thought, should this be `types.enum [ "ssh" "ssh-ng" ]` instead?

Comment (yajo) [reply nixos/modules/services/misc/nix-daemon.nix:232]: Ah, I didn't know about it.

File: nixos/modules/services/misc/nix-daemon.nix [hunk 2]
```diff
@@ -228,6 +228,16 @@ in
                 The hostname of the build machine.
               '';
             };
+            protocol = mkOption {
+              type = types.enum [ "ssh" "ssh-ng" ];
+              default = "ssh";
+              example = "ssh-ng";
+              description = ''
```
Comment (SuperSandro2000) [inline nixos/modules/services/misc/nix-daemon.nix:235]:
```suggestion
              description = lib.mdDoc ''
```

File: nixos/modules/services/misc/nix-daemon.nix [hunk 3]
```diff
@@ -228,6 +228,16 @@ in
                 The hostname of the build machine.
               '';
             };
+            protocol = mkOption {
+              type = types.enum [ "ssh" "ssh-ng" ];
+              default = "ssh";
+              example = "ssh-ng";
+              description = lib.mdDoc ''
+                The protocol used for communicating with the build machine.
+                Use <literal>ssh-ng</literal> if your remote builder and your
```
Comment (SuperSandro2000) [inline nixos/modules/services/misc/nix-daemon.nix:237]:
```suggestion
                Use `ssh-ng` if your remote builder and your
```

**Comment by SuperSandro2000** [comment]
succeeded by #191713

**Pull request closed without merging**